include/RAJA/util/Layout.hpp

```diff
@@ -166,6 +167,28 @@ struct LayoutBase_impl<camp::idx_seq<RangeInts...>, IdxLin, StrideOneDim> {
   {
   }
 
+  /*!
+   * Methods to performs bounds checking in layout objects
+   */
+  template<camp::idx_t N, typename Idx>
+  RAJA_INLINE RAJA_HOST_DEVICE void BoundsCheckError(Idx idx) const
+  {
+    printf("View index %ld is out of bounds [0, %ld) \n" ,static_cast<long int>(idx), static_cast<long int>(sizes[N]));
```

Comment:
Two things:  this is a Layout not a View so the comment should reflect that (people use Layouts without Views... or at least I do), and how does this work with Layouts that don't have a [0,N) range? such as the OffsetLayout?

Reply:
That's a good point, I'll fix the comment. Offsets layouts actually go through regular views to do the index calculation. The offset layouts just have an additional step where they have to remove the "offsets". I can do something similar for the offset layouts and intercept right before the offsets are removed to get more accurate information about the interval range.

Reply:
Done.